### DISCUSSION TITLE
[New Feature] "Hold Ground" Command

### DISCUSSION THREAD

**Discussion by CCTiberius** [Vinifera-Developers/Vinifera#1496] [Feature Requests]
### Description:
There being no way to completely stop a unit from moving unless told to is just annoying, and makes using certain units frustrating, especially defensively.

### Possible Implementation:
A customizable hotkey similar to "guard", but which tells the unit to stand still *no matter what* until it is given a move order by the player. They would fire on enemies that come within range, but would never ever move to reach an enemy.

**Comment by Rampastring** [comment]
A stance system like in C&C3 would be nice.

**Comment by Bittah** [comment]
The way units behave when it comes to chasing after enemy units or returning back to their initial position is pretty messed up in general in TS and it works much better in YR.

It'd be much better if units in Guard mode never move from their position by default to begin with _unless_ they're fired upon first and when a unit _does_ chase after an enemy unit, it should give up and return back to its initial position when the enemy unit has moved too far out of range (just like units do in YR).

**Comment by Onidotmoe** [comment]
> A stance system like in C&C3 would be nice.

Yes! with a GUI in the bottom left corner!
Would also be super useful to have a do-not-attack stance like in C&C3 for stealth units to not reveal their location.

**Comment by hacklex** [comment]
What would happen if we just make a hotkey to assign to all selected units `Mission=Sticky`? How difficult is that to implement?

**Comment by JoyfulShush** [comment]
I think this is a great QoL. Can we discuss what would be good ways to do this? `Mission=sticky` might be the obvious choice (makes units never move, but attack within range), though it seems to come with its own set of issues and hardcoded behaviors.

Would a new mission be a good idea for that?
